internal/transport/transport_test.go

```diff
@@ -2959,3 +2959,21 @@ func (s) TestReadHeaderMultipleBuffers(t *testing.T) {
 		t.Errorf("bytesRead = %d, want = %d", bytesRead, headerLen)
 	}
 }
+
+func TestSetRstAndCallNoop(t *testing.T) {
+	ch := make(chan struct{}, 1)
+	noop = func() {
+		ch <- struct{}{}
+	}
+	var rst bool
+	setRstAndCallNoop(&rst)
```

Comment:
this has to be declared in the test. Please see resolver tests how things are being done there. You need to also assign it back to original definition